sinks/cortex/cortex_test.go

```diff
@@ -281,10 +289,12 @@ func TestChunkedWritesRespectContextCancellation(t *testing.T) {
 	// Perform the flush to the test server
 	flushResult, err := sink.Flush(ctx, metrics)
 	assert.Error(t, err)
-	assert.Equal(t, sinks.MetricFlushResult{MetricsFlushed: 4, MetricsDropped: 3, MetricsSkipped: 0}, flushResult)
+	assert.Equal(t, sinks.MetricFlushResult{MetricsFlushed: 3, MetricsDropped: 9, MetricsSkipped: 0}, flushResult)
 
 	// we're cancelling after 2 so we should only see 2 chunks written
 	assert.Equal(t, 2, len(server.History()))
+	assert.Equal(t, 3, len(server.History()[0].data.GetTimeseries()))
+	assert.Equal(t, 3, len(server.History()[1].data.GetTimeseries()))
```

Comment:
Wanted to verify expected behavior here:
There are 12 total samples in the chunked_input.json. The write_batch_size is 3:

1. We write 3 successfully.
2. We write 3 more, but context gets canceled. I think there's some ambiguity about whether the write finished or was interrupted. We consider these 3 as dropped. (Q: Are we ok with this behavior? These 3 will be considered dropped, despite them possibly having already gotten forwarded and processed.)
3. The last 6 are not written at all. I've added these under dropped. (Q: Should these be considered dropped? Or skipped?)

Reply:
For Qs:
1. Anytime we can't get a hard confirmation metrics have been written we should consider them dropped. If/when a retry mechanism is implemented sending these "dropped" metrics again would do nothing, they'd just be rejected for already being written. 

2. Dropped, we didn't intentionally skip over the metrics (like, say, we filtered out metrics that we can't send anything for w/e reason) but dropped them on the floor because we took too long. 

All skipped metrics are dropped metrics but I consider 'dropped metrics' a result of an error and skipped metrics are intentional (and I'd call any metrics not submitted after the context deadline an error scenario)

Reply:
> We will consider these 3 as dropped. (Q: Are we ok with this behavior? These 3 will be considered dropped, despite them possibly having already gotten forwarded and processed. In the test case, they DO get forwarded successfully, but context is expired before the client gets a response and can bookkeep properly.)

Correct, these should be considered dropped. We cannot confirm success therefore they are dropped.

> The last 6 are not written at all. I've added these under dropped. (Q: Should these be considered dropped? Or skipped?)

Dropped should be appropriate here for the same reason